earth_enterprise/src/common/CacheSizeCalculations.h

```diff
@@ -19,29 +19,77 @@
 #include "khTypes.h"
 #include <memory>
 #include <vector>
+#include <set>
+#include <map>
+#include <string>
+#include "notify.h"
+#include <qstring.h>
+#include <typeinfo>
+#include <type_traits>
+#include "SharedString.h"
 
-// determine amount of memory used by generic objects
+//determine amount of memory used by an object pointed to by a pointer
 template<class T>
-inline uint64 GetObjectSize(const T &obj) {
-    return sizeof(obj);
+inline uint64 GetHeapUsage(const T* &ptr) {
+    uint64 ptrSize = (ptr ? GetSize(*ptr) : 0);
```

Comment:
This apparently works because it compiles, but I'm confused at how you're able to call GetSize in this function when GetSize is defined below. It seems like you would need a function declaration above this. Is there a chance that it's somehow calling the wrong function, or have you already verified that?

Reply:
I hadn't thought of that, but I have verified it works. I'll move it for peace of mind.

Reply:
If you move it you have the opposite problem because GetSize calls GetHeapUsage. I think the only "safe" path is to put just a function declaration at the top for GetSize - then everything is covered.